### ISSUE TITLE
Left Join & 'as' option

### ISSUE THREAD

**Issue by olintononthemovesw** [ujjwalguptaofficial/JsStore#124]
When performing a left join, there is an error when applying an 'as' option
Example (https://ujjwalguptaofficial.github.io/idbstudio/?db=Demo)
Initialise with:
```
update({
    in: "Orders",
    set: {
        shipperId: 1234567890
    },
    where: {
        orderId: 10248
    }
});
```
Then:
```
select({
    from: "Orders",
    join: {
        with: "Shippers",
        on: "Shippers.shipperId = Orders.shipperId",
        type: "left"
    }
});
```
will correctly return the the records...
![image](https://user-images.githubusercontent.com/54812461/64359296-f6753f80-cfff-11e9-9e09-ae9c1b97f66e.png)

But...
```
select({
    from: "Orders",
    join: {
        with: "Shippers",
        on: "Shippers.shipperId = Orders.shipperId",
        as: {
            shipperId: "Shippers.shipperId",
            shipperName: "Shippers.shipperName",
            phone: "Shippers.phone"
        },
        type: "left"
    }
});
```
Will not apply the alternative field names, as well as resulting in undefined field values.
![image](https://user-images.githubusercontent.com/54812461/64359380-1b69b280-d000-11e9-8478-dd4152fe0558.png)

**Comment by ujjwalguptaofficial** [comment]
Issue fixed in v - 3.3.6.

**Comment by olintononthemovesw** [comment]
Tested. For the most parts this is now working. Thank you!

We have however found an edge case when applying an 'as' statement...
https://github.com/ujjwalguptaofficial/JsStore/issues/125
